python/ray/data/datasource/torch_datasource.py

```diff
@@ -0,0 +1,62 @@
+from typing import Callable, List
+import logging
+
+import torch.utils.data
+
+from ray.data.block import Block, BlockMetadata, T
+from ray.data.datasource import Datasource, ReadTask
+
+logger = logging.getLogger(__name__)
+
+
+class TorchDatasource(Datasource[T]):
+    """A datasource that let's you use Torch datasets with Ray Data.
+
+    .. warning::
+        ``TorchDatasource`` doesn't support parallel reads. You should only use this
+        datasource for small datasets like MNIST or CIFAR.
+
+    Example:
+        >>> dataset_factory = lambda: torchvision.datasets.MNIST("data", download=True)
+        >>> dataset = ray.data.read_datasource(
+                TorchDatasource(), parallelism=1, dataset_factory=dataset_factory
```

Comment:
Should dataset_factory be passed as a constructor arg to TorchDatasource?

Reply:
Discussed with @clarkzinzow, and we decided to place `dataset_factory` in `prepare_read` to be consistent with the other datasources.